handler.js

```diff
@@ -1,26 +1,66 @@
 'use strict';
 
+var http = require('http')
 var https = require('https')
 
+var makeRequest = (protocol, host, path, callback) => {
+  var req = protocol.get({ host, path }, (res) => {
+    let body = ''
+
+    res.on('data', (d) => {
+      if (connectTimeout) { clearTimeout(connectTimeout) }
+      body += d
+    })
+    res.on('end', () => {
+      try {
+        callback(null, JSON.parse(body))
+      } catch (e) {
+        callback('json parse error')
+      }
+    })
+  })
+
+  req.on('socket', (socket) => {
+    socket.setTimeout(5000)
+    socket.on('timeout', () => {
+      // console.log('socket timeout');
+      req.abort()
+    })
+  })
+
+  const connectTimeout = setTimeout(() => {
+    // console.log('connect timeout')
+    req.abort()
+  }, 5000)
+
+  req.on('error', callback)
+
+  req.end()
+}
+
+var sanitizeChars = (s) => { return s.replace(/[^ a-zA-Z]/g, '?') }
+
 // Your first function handler
 module.exports.trevorbot = (event, context, cb) => {
-  if (event.body.text.toLowerCase().indexOf('where') > -1) {
-    if (event.body.text.toLowerCase().replace('trevorbot', '').indexOf('trevor') > -1 ||
-      event.body.user_name.toLowerCase().indexOf('trevor') > -1 && event.body.text.toLowerCase().indexOf(' i') > -1) {
-      https.get({ host: 'nomadlist.com', path: '/trevorgerhardt.json'}, (res) => {
-        let body = ''
-
-        res.on('data', (d) => body += d)
-        res.on('end', () => {
-          let data = JSON.parse(body)
-          cb(null, { text: `Trevor is in ${data.location.now.city.replace(/[^ a-zA-Z]/g, '?')}, ${data.location.now.country.replace(/[^ a-zA-Z]/g, '?')} (https://nomadlist.com/trevorgerhardt)` })
-        })
+  var text = event.body.text.toLowerCase()
+  if (text.indexOf('where') > -1) {
+    if (text.replace('trevorbot', '').indexOf('trevor') > -1 ||
+      event.body.user_name.toLowerCase().indexOf('trevor') > -1 && text.indexOf(' i') > -1) {
+      makeRequest(https, 'nomadlist.com', '/trevorgerhardt.json', (err, data) => {
+        if (err) { return cb(null, { text: 'I couldn\'t figure that out right now :astonished:' }) }
+        var city = sanitizeChars(data.location.now.city)
+        var country = sanitizeChars(data.location.now.country)
+        cb(null, { text: `Trevor is in ${city}, ${country} (https://nomadlist.com/trevorgerhardt)` })
       })
     } else {
       cb(null, { text: 'I only know where Trevor is.' })
     }
+  } else if (text.indexOf('chuck norris') > -1) {
+    makeRequest(http, 'api.icndb.com', '/jokes/random', (err, data) => {
+      if (err) { return cb(null, { text: 'I don\'t feel like doing that right now :pensive:' }) }
+      cb(null, { text: data.value.joke })
```

Comment:
You could expand the regex to include numbers and punctuation. It's to avoid having slack formatting passed through.